src/google/protobuf/compiler/objectivec/objectivec_generator.cc

```diff
@@ -190,19 +190,21 @@ bool ObjectiveCGenerator::GenerateAll(
       // header search path since the generate #import will be more complete.
       generation_options.runtime_import_prefix =
           StripSuffixString(options[i].second, "/");
-    } else if (options[i].first == "default_objc_class_prefix") {
-      // The default objc class prefix to use if specified by the command line 
-      // invocation. The file option is always honored first if one is present.
-      std::string value = options[i].second;
-      if (value.empty()) {
-        *error = "error: default_objc_class_prefix cannot be empty.";
-        return false;
-      }
-      SetDefaultObjcClassPrefix(value);
+    } else if (options[i].first == "objc_class_prefix_mappings_path") {
+      // Path to use for when loading the objc class prefix mappings to use.
+      // The `objc_class_prefix` option is always honored first if one is present.
+      //
+      // The format of the file is:
```

Comment:
Since you are keying this by files, it means you are explicitly allowing a proto package to be split between multiple prefixes, which is the opposite direction from what the library was generally trying to encourage with the prefix validation support.  Is there a reason you went this directions vs. trying to keep package on a single prefix?

Reply:
Our problem is that we use sub-directories to organize protos and in some cases the proto filenames are the same.

e.g. company/featureA/service.proto and company/featureB/service.proto.

The .proto files have a different `package` declaration to separate those but using a single/global prefix causes duplicate class names to be generated. So we generate our prefix based on the `package` declared in order to avoid duplicates.

Reply:
Sorry, I'm not following.  I was saying you'd use a file that is the same as the prefix validation one, so _package_ = _prefix_ lines.  For what you describe, that sounds like it would work just fine.

If the two files were in the _same_ proto package, then it wouldn't work.

Reply:
ah I see what you are saying yes this seems like it would for our case as well. I can update this.

Reply:
done!